src/utilities.jl

```diff
@@ -12,6 +12,41 @@ function _get_latest_version(base_path::AbstractString)
     end
 end
 
+"""
+    _find_latest_pkg_entry(pkg_name::Union{AbstractString, Missing}, pkg_uuid::Union{Missing, UUID}=missing; registries::Array{RegistryInstance}=reachable_registries())
+
+Given a list of registries, find the entry corresponding to the given package (specified as a `name` and/or an `UUID`) with the latest version.
+
+For example, if a package is registered in two registries with the same name and UUID, returns the `PkgEntry` corresponding to the registry which
+has the higher version number of the package.
+"""
+function _find_latest_pkg_entry(pkg_name::Union{AbstractString, Missing}, pkg_uuid::Union{Missing, UUID}=missing; registries::Array{RegistryInstance}=reachable_registries())
+    if ismissing(pkg_name) && ismissing(pkg_uuid)
+        throw(ArgumentError("Must supply either a `pkg_name` or `pkg_uuid`!"))
+    end
+    entries = PkgEntry[]
+    for rego in registries
+        for (name, entry) in rego.pkgs
+            if !ismissing(pkg_name)
+                pkg_name == name || continue
+            end
+            if !ismissing(pkg_uuid)
+                pkg_uuid == entry.uuid || continue
+            end
```

Comment:
I feel like this can be simplified, haven't given a thought too much about it yet

Reply:
Yeah, I wasn't totally happy with that but it seemed tricky to get the right logic of "if `pkg_name` is provided it must be correct, but if it's not, then skip the check" (and same for `pkg_uuid`) with shorter ways. I thought at least it's fairly explicit this way.